// @rxweb/reactive-form-validators/esm2015/directives/control-host.directive.js
import * as tslib_1 from "tslib";
import { EmbeddedViewRef, Directive, Input, ViewContainerRef, OnDestroy } from '@angular/core';
let ControlHostDirective = class ControlHostDirective {
    constructor(viewContainerRef) {
        this.viewContainerRef = viewContainerRef;
    }
    set portal(context) {
        if (context.templateRef) {
            if (this.view) {
                this.view.destroy();
                this.view = undefined;
            }
            this.view = this.viewContainerRef.createEmbeddedView(context.templateRef, context);
        }
    }
    ngOnDestroy() {
        if (this.view)
            this.view.destroy();
        if (this.viewContainerRef)
            this.viewContainerRef.clear();
    }
};
ControlHostDirective.ctorParameters = () => [
    { type: ViewContainerRef }
];
tslib_1.__decorate([
    Input('controlHost')
], ControlHostDirective.prototype, "portal", null);
ControlHostDirective = tslib_1.__decorate([
    Directive({
        selector: '[controlHost]'
    })
], ControlHostDirective);
export { ControlHostDirective };
